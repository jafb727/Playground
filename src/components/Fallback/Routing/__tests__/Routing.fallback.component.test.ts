/**
 * @author: Jose A Felix
 * @editor: Jose A Felix
 * @name: Routing.fallback.component.test
 * @description: Routing fallback component test cases
 */

/* --------------------------------------------- */

/** @import Libraries */
import { screen } from "@testing-library/react";

/** @import Components */
import Component from "../Routing.fallback.component";

/** @import Utilities */
import { routerSetup, history } from "../../../../test/setupTests";
import userEvent from "@testing-library/user-event";

/* --------------------------------------------- */

/** @event Mocking events */
window.HTMLMediaElement.prototype.load = () => {};

/* --------------------------------------------- */

describe("Routing.fallback.component", () => {
   /** @constant Routing setup */
   const routingSetup = {
      location: "/error",
      path: "/error",
   };

   /* ----------------------- */

   test("Renders component ok", () => {
      routerSetup(Component, routingSetup);
      expect(screen.getByTestId("routing-fallback")).toBeInTheDocument();
   });

   /* ----------------------- */

   test("Back home button navigates ok", async () => {
      routerSetup(Component, routingSetup);
      const button = screen.getByTestId("back-home-button");
      await userEvent.click(button);
      expect(history.location.pathname).toBe("/");
   });

   /* ----------------------- */

   test("Passes shallow snapshot test", () => {
      const component = routerSetup(Component, routingSetup);
      expect(component).toMatchSnapshot();
   });
});
