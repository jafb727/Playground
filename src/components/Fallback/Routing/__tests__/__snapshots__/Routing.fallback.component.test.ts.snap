// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Routing.fallback.component Passes shallow snapshot test 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiBox-root css-0"
        data-testid="routing-fallback"
      >
        <div
          class="fullPageBackground defaultBackground undefined MuiBox-root css-0"
          data-testid="background"
        >
          <div
            class="frameContainer MuiBox-root css-0"
          >
            <video
              autoplay=""
              class="spaceVideoBackground fade"
              data-testid="space-video-background"
              loop=""
              poster=""
            >
              <source
                src=""
                type="video/mp4"
              />
              <p
                class="MuiTypography-root MuiTypography-body1 css-10kbhik-MuiTypography-root"
              >
                Your browser does not support HTML5 video
              </p>
            </video>
            <div
              class="moonObject MuiBox-root css-0"
            >
              <img
                alt="Moon"
                src="MoonAstronaut.png"
              />
            </div>
            <div
              class="errorMessage MuiBox-root css-0"
              data-testid="error-message"
            >
              <span
                class="MuiTypography-root MuiTypography-h1_1 MuiTypography-gutterBottom errorHeader css-kjxf0h-MuiTypography-root"
              >
                404
              </span>
              <h4
                class="MuiTypography-root MuiTypography-h4 MuiTypography-gutterBottom errorSubtitle css-1m6o9zw-MuiTypography-root"
              >
                Uh oh! You're lost
              </h4>
              <p
                class="MuiTypography-root MuiTypography-body1 errorApology css-10kbhik-MuiTypography-root"
              >
                The page you are looking for does not exist. How you got here is a mystery. But you can click the button below to go back to the homepage
              </p>
              <a
                href="/"
              >
                <button
                  class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium backHomeButton css-5y2byu-MuiButtonBase-root-MuiButton-root"
                  data-testid="back-home-button"
                  tabindex="0"
                  type="button"
                >
                  <p
                    class="MuiTypography-root MuiTypography-body1 css-10kbhik-MuiTypography-root"
                  >
                    Back Home
                  </p>
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </button>
              </a>
            </div>
            <footer
              class="footerFullPage"
            >
              <div
                class="collectionContainer undefined MuiBox-root css-1a9getn"
                data-testid="collection"
              >
                <div
                  class="copyright MuiBox-root css-0"
                  data-testid="copyright-component"
                >
                  <p
                    class="MuiTypography-root MuiTypography-body2 css-n6q3br-MuiTypography-root"
                  >
                    © 
                    All rights reserved.
                  </p>
                </div>
              </div>
            </footer>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiBox-root css-0"
      data-testid="routing-fallback"
    >
      <div
        class="fullPageBackground defaultBackground undefined MuiBox-root css-0"
        data-testid="background"
      >
        <div
          class="frameContainer MuiBox-root css-0"
        >
          <video
            autoplay=""
            class="spaceVideoBackground fade"
            data-testid="space-video-background"
            loop=""
            poster=""
          >
            <source
              src=""
              type="video/mp4"
            />
            <p
              class="MuiTypography-root MuiTypography-body1 css-10kbhik-MuiTypography-root"
            >
              Your browser does not support HTML5 video
            </p>
          </video>
          <div
            class="moonObject MuiBox-root css-0"
          >
            <img
              alt="Moon"
              src="MoonAstronaut.png"
            />
          </div>
          <div
            class="errorMessage MuiBox-root css-0"
            data-testid="error-message"
          >
            <span
              class="MuiTypography-root MuiTypography-h1_1 MuiTypography-gutterBottom errorHeader css-kjxf0h-MuiTypography-root"
            >
              404
            </span>
            <h4
              class="MuiTypography-root MuiTypography-h4 MuiTypography-gutterBottom errorSubtitle css-1m6o9zw-MuiTypography-root"
            >
              Uh oh! You're lost
            </h4>
            <p
              class="MuiTypography-root MuiTypography-body1 errorApology css-10kbhik-MuiTypography-root"
            >
              The page you are looking for does not exist. How you got here is a mystery. But you can click the button below to go back to the homepage
            </p>
            <a
              href="/"
            >
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium backHomeButton css-5y2byu-MuiButtonBase-root-MuiButton-root"
                data-testid="back-home-button"
                tabindex="0"
                type="button"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 css-10kbhik-MuiTypography-root"
                >
                  Back Home
                </p>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </a>
          </div>
          <footer
            class="footerFullPage"
          >
            <div
              class="collectionContainer undefined MuiBox-root css-1a9getn"
              data-testid="collection"
            >
              <div
                class="copyright MuiBox-root css-0"
                data-testid="copyright-component"
              >
                <p
                  class="MuiTypography-root MuiTypography-body2 css-n6q3br-MuiTypography-root"
                >
                  © 
                  All rights reserved.
                </p>
              </div>
            </div>
          </footer>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
